crates/sui-core/src/state_accumulator.rs

```diff
@@ -43,18 +44,47 @@ impl StateAccumulator {
 
         let mut acc = Accumulator::default();
 
+        // process insertions to the set
         acc.insert_all(
             effects
                 .iter()
                 .flat_map(|fx| {
                     fx.created
                         .clone()
                         .into_iter()
-                        .map(|(obj_ref, _)| obj_ref.2)
+                        .map(|(oref, _)| oref.2)
+                        .chain(fx.wrapped.clone().into_iter().map(|oref| oref.2))
```

Comment:
wrapped objects are not an insert are they?

Reply:
When we wrap an object, we create a new object (the parent) which has a field that refers to the child. The child goes away (is deleted). My understanding is that means we need to add all wraps and unwraps to `inserted` (likewise, my understanding was `unwrap` creates the original object and deletes the wrapped object)

Reply:
Actually after saying this out loud, I think I will need to process `modified_at_versions` for deletions as well, since (I think) wraps and unwraps create a deleted object or wraped obejct, respectively

Reply:
@amnn can you comment?

Reply:
I recall William and I talked about this at Vegas, and at the time, settled on the fact that wrapped and unwrapped objects were not exactly the same as deletions and insertions for your purposes.

IIUC, this information is being used to roll-up the live set of objects in the database so that a node can bulk load them into its own database to catch up.  For those purposes, a wrapped object shouldn't be in the live set, because it's not accessible at its ID, but it probably shouldn't be treated as deleted either, because it needs to be present in the DB as wrapped so that a later unwrap can re-inhabit it.

Reply:
Also note that you do not always need to create the parent object when wrapping -- it could have existed prior to the wrap -- and similarly, you do not need to destroy the parent to unwrap the child either (imagine a child that was wrapped in a vector and then later popped off that vector).

Reply:
Isn't this a problem then if wraps don't change anything from the perspective of the hash? A node could generate a snapshot with objects unwrapped (e.g. staked tokens) that should be wrapped as an attack vector. What is preventing this? @mystenmark @amnn

Reply:
How would this be different than a node inserting arbitrary objects that shouldn't be there?